docs/source/get_started/implementers_best_practices.rst

```diff
@@ -0,0 +1,135 @@
+============================
+Implementers' best practices
+============================
+
+.. contents:: Table of Contents
+   :depth: 2
+
+The goal of this document
+=========================
+
+The goal of this document is to provide a set of tips, instructions and good practices for implementations built on
+top of the MOTECH platform or using MOTECH in the backend. The content of this document has been built basing on
+our experiences with the implementations and the MOTECH platform.
+
+
+Hardware recommendations
+========================
+
+The hardware requirements will vary, depending on the amount and type of requests that your instance will serve at the same time
+and the amount of data, that you plan to store. This includes the installed bundles, log file and the database. We
+recommend that the instance running the MOTECH server has got at least 4GB of RAM. Lower amounts may considerably
+slow down the processing.
+
+Of course the best way to find out the optimal configuration is to perform some stress testing. There are a lot of tools
+that can simulate the traffic to the server. Running such stress tools on servers with various configurations will help to
+determine at which point adding more resources stops adding up to the amount of processed requests.
+
+
+Topology recommendations
+========================
+
+MOTECH can be run in a clustered environment. This means that you can run several MOTECH instances in order to distribute
+the workload across multiple processing units. Similarly, you can set up multiple ActiveMQ instances that handle MOTECH
+events. In a topology with several MOTECH instances, you can choose whether to have the event handled by one instance
+only or, what is a rare case, by all instances. The second option is usually only used to notify the instances about some
+changes that must be immediately reflected on the instances. Finally, it is also possible to have clustered databases.
+There are two approaches when it comes to clustering databases. In the master-master model, all of the databases can update
+the data and the changes get replicated across all of the instances. In the master-slave model, only one of the databases
+(considered master) handles data updates and pushes the updated data to the slave servers. Of course using clustered
+environment is not a requirement. You can use single MOTECH, ActiveMQ and database instance, depending on the expected
```

Comment:
A reason for using multiple instances, ActiveMQ in particular is failover, meaning one server going down bring down the system. I think this section could use some mention of that.